src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -263,6 +263,11 @@ public override HttpRequestMessage CreateInvokeMethodRequest(HttpMethod httpMeth
                 }
             };
 
+            if (this.apiTokenHeader is not null)
+            {
+                request.Headers.TryAddWithoutValidation(this.apiTokenHeader.Value.Key, this.apiTokenHeader.Value.Value);
+            }
```

Comment:
This was a bug that no one had reported yet, but worth patching. Found this while reproducing the bug.